### PR TITLE
Record the TypeKind of arithmetic operation in ArithmeticVariableSlot

### PR DESCRIPTION
**Pull request by d367wang**
Record the TypeKind of the arithmetic operation result, which indicates the value range.
The result depends on the wider operand. i.e.

- If both sides are int or narrower than int => arithmetic slot's type kind int

- If one side is long => arithmetic slot's type kind is long

### CODE REVIEW

File: src/checkers/inference/model/ArithmeticVariableSlot.java [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
  * {@link Slot}s. Note that this slot is serialized identically to a {@link Slot}.
  */
 public class ArithmeticVariableSlot extends VariableSlot {
+    /**
+     * The value kind of the arithmetic operation, which indicates the value range.
+     * The result depends on the wider operand. i.e.
+     * If both sides are narrower than int => int
+     * If one sides is long => long
+     */
+    private TypeKind valueTypeKind;
```
Comment (wmdietl) [inline src/checkers/inference/model/ArithmeticVariableSlot.java:16]:
```suggestion
    private final TypeKind valueTypeKind;
```

File: src/checkers/inference/DefaultSlotManager.java [hunk 2]
```diff
@@ -418,16 +421,37 @@ public ExistentialVariableSlot createExistentialVariableSlot(Slot potentialSlot,
...
+                || TypeKindUtils.primitiveOrBoxedToTypeKind(rhsType) == TypeKind.LONG) {
+            return TypeKind.LONG;
+        }
+
+        return TypeKind.INT;
+    }
+
     @Override
-    public ArithmeticVariableSlot createArithmeticVariableSlot(AnnotationLocation location) {
+    public ArithmeticVariableSlot createArithmeticVariableSlot(
```
Comment (d367wang) [inline src/checkers/inference/DefaultSlotManager.java:444]: javadoc

File: src/checkers/inference/DefaultSlotManager.java [hunk 3]
```diff
@@ -418,16 +421,37 @@ public ExistentialVariableSlot createExistentialVariableSlot(Slot potentialSlot,
         return existentialVariableSlot;
     }
 
+
```
Comment (d367wang) [inline src/checkers/inference/DefaultSlotManager.java:424]: reference to JLS

File: src/checkers/inference/SlotManager.java [hunk 4]
```diff
@@ -150,7 +150,8 @@
      * @param location an AnnotationLocation used to locate this variable in code
      * @return the ArithmeticVariableSlot for the given location
      */
-    ArithmeticVariableSlot createArithmeticVariableSlot(AnnotationLocation location);
+    ArithmeticVariableSlot createArithmeticVariableSlot(
```
Comment (d367wang) [inline src/checkers/inference/SlotManager.java:153]: comments on new parameters

File: src/checkers/inference/model/ArithmeticVariableSlot.java [hunk 5]
```diff
@@ -1,13 +1,23 @@
...
 
-    public ArithmeticVariableSlot(int id, AnnotationLocation location) {
+    public ArithmeticVariableSlot(int id, AnnotationLocation location, TypeKind valueTypeKind) {
```
Comment (d367wang) [inline src/checkers/inference/model/ArithmeticVariableSlot.java:18]: javadoc

**Pull request merged**